packages/frontend/@n8n/design-system/src/v2/components/DropdownMenu/DropdownMenuItem.vue

```diff
@@ -0,0 +1,375 @@
+				</DropdownMenuSubContent>
+			</DropdownMenuPortal>
+		</DropdownMenuSub>
+
+		<!-- Regular item without children -->
+		<DropdownMenuItem
+			v-else
+			:disabled="disabled"
+			:class="[$style.item, props.class, { [$style.highlighted]: highlighted }]"
+			@select="handleItemSelect"
+		>
+			<slot name="item-leading" :item="props" :ui="leadingProps">
+				<Icon
+					v-if="icon?.type === 'icon'"
+					:icon="icon.value"
+					:class="[$style['item-leading'], $style.icon]"
+					size="large"
+					:color="disabled ? 'text-xlight' : 'text-light'"
+				/>
+				<span v-else-if="icon?.type === 'emoji'" :class="[$style['item-leading'], $style.emoji]">
+					{{ icon.value }}
+				</span>
+			</slot>
+			<N8nText
+				:class="$style['item-label']"
+				size="medium"
+				:color="disabled ? 'text-light' : 'text-dark'"
+			>
+				{{ label }}
+			</N8nText>
+			<slot name="item-trailing" :item="props" :ui="trailingProps" />
+			<Icon
+				v-if="checked"
+				icon="check"
+				:class="$style['item-check']"
+				size="large"
+				:color="disabled ? 'text-xlight' : 'text-light'"
+			/>
+		</DropdownMenuItem>
+	</div>
+</template>
+
+<style module>
+.wrapper {
+	display: contents;
+}
+
+.children-container {
+	padding: var(--spacing--4xs);
+}
+
+.item {
+	font-size: var(--font-size--2xs);
+	line-height: 1;
+	border-radius: var(--radius);
+	display: flex;
+	align-items: center;
+	min-height: var(--spacing--lg);
+	padding: var(--spacing--2xs);
+	position: relative;
+	user-select: none;
+	color: var(--color--text--shade-1);
+	gap: var(--spacing--3xs);
+	outline: none;
+
+	&:not([data-disabled]) {
+		&:hover,
+		&[data-highlighted],
+		&.highlighted {
+			background-color: var(--color--background--light-1);
+			cursor: pointer;
+		}
+	}
+
+	&[data-disabled] {
+		color: var(--color--text--tint-1);
+		cursor: not-allowed;
+	}
+}
+
+.sub-trigger {
+	&[data-state='open'] {
+		background-color: var(--color--background--light-1);
+	}
+}
+
+.sub-indicator {
+	margin-left: auto;
+	flex-shrink: 0;
+	color: var(--color--text--tint-1);
+}
+
+.sub-content {
+	min-width: 160px;
+	border-radius: var(--radius);
+	border: var(--border);
+	background-color: var(--color--background--light-2);
+	box-shadow: var(--shadow);
+	z-index: 999999;
+}
+
+.item-leading {
+	flex-shrink: 0;
+}
+
+.emoji {
+	font-size: var(--font-size--xs);
+	line-height: 1;
+}
+
+.item-label {
+	flex-grow: 1;
+}
+
+.item-check,
+.item-trailing {
+	margin-left: auto;
+	flex-shrink: 0;
+}
+
+.separator {
+	height: 1px;
+	background-color: var(--color--foreground);
+	margin: var(--spacing--4xs) 0;
+}
+
+.loading-container {
+	padding: var(--spacing--4xs);
+}
+
+.loading-item {
+	margin-bottom: var(--spacing--4xs);
+
+	&:last-child {
+		margin-bottom: 0;
+	}
+}
+
+.empty-state {
+	padding: var(--spacing--2xs) var(--spacing--xs);
+	color: var(--color--text--tint-1);
+	font-size: var(--font-size--sm);
+	text-align: center;
+}
```

Comment:
Empty state isn't spaced properly. That text should be vertically centered between the bottom and the search border (though maybe addressing the padding and height issues fix it?)

<img width="308" height="156" alt="Image" src="https://github.com/user-attachments/assets/11499b52-b915-4eb9-bbe3-e4948fdb785a" />